### PR TITLE
Log a warning if a coremod does not follow best practices

### PR DESCRIPTION
**Pull request by ichttt**
This will (hopefully) make modders aware of the new coremod rules as pointed out [here](http://www.minecraftforge.net/forum/topic/58706-regarding-minecraft-112-and-policy-changes/).

### CODE REVIEW

File: src/main/java/net/minecraftforge/fml/relauncher/CoreModManager.java [hunk 1]
```diff
@@ -595,6 +597,32 @@ else if (requiredMCVersion != null)
...
+                    FMLLog.log.info("Ignoring missing certificate for coremod {} ({}), we are in deobf and it's a forge core plugin", coreModName, coreModClass);
+                }
+                else if (deobfuscatedEnvironment && location == null) // This is probably a mod dev workspace, don't crash in 1.13
+                {
+                    FMLLog.log.info("The coremod {} ({}) is not signed, but this is a probably dev workspace - ignoring", coreModName, coreModClass);
+                }
+                else // TODO This is a probably a normal minecraft workspace - crash if no signature data is found in 1.13
+                {
+                    FMLLog.bigWarning("The coremod {} ({}) is not signed! Coremods should be to be signed, and this will required in 1.13", coreModName, coreModClass);
+                }
```
Comment (williewillus) [inline src/main/java/net/minecraftforge/fml/relauncher/CoreModManager.java]: "should be to be"?

Comment (ichttt) [reply src/main/java/net/minecraftforge/fml/relauncher/CoreModManager.java]: woops, should read "should be signed"

File: src/main/java/net/minecraftforge/fml/relauncher/CoreModManager.java [hunk 2]
```diff
@@ -595,6 +597,32 @@ else if (requiredMCVersion != null)
...
+                {
+                    FMLLog.log.info("Ignoring missing certificate for coremod {} ({}), we are in deobf and it's a forge core plugin", coreModName, coreModClass);
+                }
+                else if (deobfuscatedEnvironment && location == null) // This is probably a mod dev workspace, don't crash in 1.13
+                {
+                    FMLLog.log.info("The coremod {} ({}) is not signed, but this is a probably dev workspace - ignoring", coreModName, coreModClass);
+                }
+                else // TODO This is a probably a normal minecraft workspace - crash if no signature data is found in 1.13
+                {
+                    FMLLog.bigWarning("The coremod {} ({}) is not signed! Coremods should be signed, and this will required in 1.13", coreModName, coreModClass);
```
Comment (liach) [inline src/main/java/net/minecraftforge/fml/relauncher/CoreModManager.java]: "will be required"

**Comment by mezz** [comment]
Those are coremod best practices, not rules. If someone wants to ignore them that's their decision.
Warnings and info are fine but we have no intention of crashing the game if coremods do not comply with best practices.

**Comment by mattmess1221** [comment]
Is it known if mods inside coremods will still be supported at a future date? If not, it would be a good idea to mention it.

**Comment by LexManos** [comment]
No, the plan is to remove that in the re-write and move to the jar-in-jar system.

File: src/main/java/net/minecraftforge/fml/relauncher/CoreModManager.java [hunk 3]
```diff
@@ -412,7 +414,7 @@ public boolean accept(File pathname)
                 }
                 else
                 {
-                    FMLLog.log.trace("Found FMLCorePluginContainsFMLMod marker in {}, it will be examined later for regular @Mod instances",
+                    FMLLog.log.warn("Found FMLCorePluginContainsFMLMod marker in {}. This is not recommended, mods should be in an other jar as the coremod.",
```
Comment (mezz) [inline src/main/java/net/minecraftforge/fml/relauncher/CoreModManager.java]: The text might be better this way:
`This is not recommended, @Mods should be in a separate jar from the coremod.`

**Review by mezz — CHANGES_REQUESTED** [review]

**Review by mezz — APPROVED** [review]

**Pull request merged**